src/functions/update-curation/handler.ts

```diff
@@ -76,40 +76,61 @@ const updateCuration: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async
   let curationResponse: CurationModel = curation;
 
   if (curationType === CurationType.CUSTOM && curation.documentId && document) {
-    const foundDocument = await documentService.findDocument(curation.documentId);
-    if (!foundDocument) {
-      return {
-        statusCode: 404,
-        body: `Document ${curation.documentId} not found`
-      };
-    }
+    try {
+      const foundDocument = await documentService.findDocument(curation.documentId);
+      if (!foundDocument) {
+        return {
+          statusCode: 404,
+          body: `Document ${curation.documentId} not found`
+        };
+      }
 
-    const { title, description, links, language } = document;
+      const { title, description, links, language } = document;
 
-    const updatesToDocument: Document = {
-      id: curation.documentId,
-      description: description,
-      language: language,
-      links: links,
-      title: title,
-      curationId: id
-    };
+      const updatesToDocument: Document = {
+        id: curation.documentId,
+        description: description,
+        language: language,
+        links: links,
+        title: title,
+        curationId: id
+      };
 
-    if (!isEqual(foundDocument, updatesToDocument)) {
-      const updatedDocument = await documentService.updateDocument(updatesToDocument);
-      if (!startTime) {
-        await elastic.updateDocuments({
-          documents: [{
-            id: curation.documentId,
-            title: title,
-            description: description,
-            links: links,
-            language: language
-          }]
-        });
+      if (!isEqual(foundDocument, updatesToDocument)) {
+        try {
+          const updatedDocument = await documentService.updateDocument(updatesToDocument);
+          if (!startTime) {
+            try {
+              await elastic.updateDocuments({
+                documents: [{
+                  id: curation.documentId,
+                  title: title,
+                  description: description,
+                  links: links,
+                  language: language
+                }]
+              });
+            } catch (error) {
+              return {
+                statusCode: 500,
+                body: `Error updating elastic document with id ${curation.documentId}, ${error}`
+              }
+            }
+
+            documentResponse = updatedDocument;
+            }
+        } catch (error) {
+          return {
+            statusCode: 500,
+            body: `Error updating document record with id ${curation.documentId}, ${error}`
+          }
+        }
```

Comment:
The reference to updatedDocument is not required in the elastic.updateDocuments() call, so the nested structure can be flattened.
Check this in the editor.
```suggestion
        try {
          documentResponse = await documentService.updateDocument(updatesToDocument);
        } catch (error) {
          return {
            statusCode: 500,
            body: `Error updating document record with id ${curation.documentId}, ${error}`
          };
        }

        if (!startTime) {
          try {
            await elastic.updateDocuments({
              documents: [{
                id: curation.documentId,
                title: title,
                description: description,
                links: links,
                language: language
              }]
            });
          } catch (error) {
            return {
              statusCode: 500,
              body: `Error updating elastic document with id ${curation.documentId}, ${error}`
            };
          }
        }
```